### ISSUE TITLE
Extracting fails when next/dynamic is used

### ISSUE THREAD

**Issue by datagutt** [lingui/js-lingui#1058]
**Describe the bug**
When using Next.js dynamic imports, Lingui is unable to extract messages from the pages/components using dynamic imports.

**To Reproduce**
Steps to reproduce the behavior, possibly with minimal code sample, e.g:

```
import { Trans } from "@lingui/react"
import dynamic from "next/dynamic"
const TranslatedComponent = dynamic(() => import("../components/TranslatedComponent"))

export default function Index() {
   return (<div><TranslatedComponent /></div>
}
```

When extracting pages/index.js, i get this error:
```Cannot process file app/pages/index.js: Cannot read property 'pagesDir' of undefined```

**Expected behavior**
Messages are extracted from pages using dynamic components.

**Additional context**

This used to work fine in Next.js 10.1, but broke in 10.2. Likely related to this: 
PR: https://github.com/vercel/next.js/pull/24281
Issue: https://github.com/vercel/next.js/issues/24566

- jsLingui version ``3.8.10``
- Babel version ``7.14.0``
- Your Babel config (e.g. `.babelrc`) or framework you use (Create React App, Meteor, etc.)
Next.JS 10.2

**Comment by semoal** [comment]
Actually crashes internally Next but extracts correctly the value to the messages file. (Tried with @canary version of Next and the latest)

Actually the fix from our side is to remove the Typescript extractor located here: https://github.com/lingui/js-lingui/blob/main/packages/cli/src/api/extractors/index.ts#L6, we already discussed this on https://github.com/lingui/js-lingui/discussions/1047 and probably we I can find some free hours I'll work hard on a next version of Lingui will all these bugs fixed.

On my head I want to add a similar API of Jest to add any extractors :)

**Comment by semoal** [comment]
Released 3.9.0 give it a try when you can, but i'm pretty sure that will fixed :)
